### PR TITLE
Handle empty endpoints list

### PR DESCRIPTION
**Pull request by dstathis**
I got a temporary error state from the loki_push_api relation when the remote unit had not yet published endpoints. That caused an exception to be raised. This PR fixes that.

### CODE REVIEW

File: tests/unit/test_log_proxy_consumer.py [hunk 1]
```diff
@@ -167,6 +167,12 @@ def test__add_client(self):
             {x["url"] for x in self.harness.charm._log_proxy._clients_list()}, expected_clients
         )
 
+    def test__empty_clients_list(self):
+        # Ensure we do not raise an exception if the clients list is empty
+        rel_id = self.harness.add_relation("log-proxy", "agent")
+        self.harness.add_relation_unit(rel_id, "agent/0")
+        self.assertEqual(self.harness.charm._log_proxy._clients_list(), [])
```
Comment (sed-i) [inline tests/unit/test_log_proxy_consumer.py:174]: Is there a more indirect way of testing this? Prefer blackbox testing over whitebox testing.
For example:
```
class TestRelatedEndpoints:
    def test_no_endpoints
    def test_one_invalid_endpoint
    def test_endpoints_without_scheme
```

**Review by sed-i — APPROVED** [review]

File: tests/unit/test_log_proxy_consumer.py (hunk 1 above)
Comment (simskij) [reply tests/unit/test_log_proxy_consumer.py:174]: I agree with Leon on this. Can we assess this using a public API instead?

**Review by simskij — DISMISSED** [review]

**Review by Abuelodelanada — DISMISSED** [review]

File: lib/charms/loki_k8s/v0/loki_push_api.py [hunk 2]
```diff
@@ -2001,7 +2001,7 @@ def _clients_list(self) -> list:
         """
         clients = []  # type: list
         for relation in self._charm.model.relations.get(self._relation_name, []):
-            clients = clients + json.loads(relation.data[relation.app]["endpoints"])
+            clients = clients + json.loads(relation.data[relation.app].get("endpoints", "[]"))
```
Comment (balbirthomas) [inline lib/charms/loki_k8s/v0/loki_push_api.py:2004]: Is this change still required ? It seems to be out of sync with current main branch. Please see [this line](https://github.com/canonical/loki-k8s-operator/blob/main/lib/charms/loki_k8s/v0/loki_push_api.py#L2011) on main branch.

Comment (mmanciop) [reply lib/charms/loki_k8s/v0/loki_push_api.py:2004]: The line you linked if bugged out big times. Passing a `""` to JSON parsing and trying to treat it as a list to concatenate to another? :D

**Comment by rbarry82** [comment]
Handled as part of #150
